### PR TITLE
insights: Migrate InsightOrgVisible and InsightTimeIntervals pings to backend

### PR DESCRIPTION
**Pull request by CristinaBirkel**
Closes https://github.com/sourcegraph/sourcegraph/issues/24481

Note: Wait to merge this until after https://github.com/sourcegraph/sourcegraph/pull/27473

## Description
The `InsightOrgVisible` ping and the `InsightTimeIntervals` ping used to be calculated by parsing the settings. With this PR they are now calculated by querying the database.

I also did a bit of refactoring, which I hope makes this easier to extend in the future:
- Broke `insight_ping_emitter.go` into 2 files: one with the functions that collect/format the aggregates and the other with the emitter itself.
- Simplified the `emit` function by breaking out separate functions for each thing we want to emit.

Side note: I'm sorry the diff is so hard to read. I probably should have done the refactor in a separate commit.

## Testing Steps
I tested this by running the SQL queries against my local database and then verifying that the pings showed the correct values here: https://sourcegraph.test:3443/site-admin/pings

I also tested out the edge case where the `InsightTimeIntervals` get regrouped due to there being some kind of overlap (for example, 30 days vs 1 month.)

### CODE REVIEW

File: internal/usagestats/code_insights_test.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func TestCodeInsightsUsageStatistics(t *testing.T) {
 
 	want.WeeklyAggregatedUsage = wantedWeeklyUsage
 	want.InsightTimeIntervals = []types.InsightTimeIntervalPing{}
-	want.InsightOrgVisible = []types.OrgVisibleInsightPing{{Type: "search"}, {Type: "lang-stats"}}
+	want.InsightOrgVisible = []types.OrgVisibleInsightPing{}
```
Comment (CristinaBirkel) [inline internal/usagestats/code_insights_test.go:103]: Were these placeholders needed for anything? I assume it's just how it happened to work. Actually, okay I just cleared my database and tested it out, and it just returns an empty array and doesn't throw any errors, so I think this is fine.

File: enterprise/internal/insights/background/pings/insights_ping_aggregators.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+const insightViewTotalCountQuery = `
+SELECT presentation_type, COUNT(*)
+FROM insight_view
+GROUP BY presentation_type;
+`
+
+const insightIntervalCountsQuery = `
+SELECT COUNT(DISTINCT(ivs.insight_view_id)), series.sample_interval_value, series.sample_interval_unit FROM insight_series AS series
+JOIN insight_view_series AS ivs ON series.id = ivs.insight_series_id
+WHERE series.sample_interval_value != 0
```
Comment (coury-clark) [inline enterprise/internal/insights/background/pings/insights_ping_aggregators.go:181]:
```suggestion
WHERE series.sample_interval_value != 0
AND series.sample_interval_value is not null
AND series.sample_interval_unit is not null
```

File: internal/usagestats/code_insights_test.go (hunk 1 above)
Comment (coury-clark) [reply internal/usagestats/code_insights_test.go:103]: I think these are the values that are were emitted by the frontend when this was implemented there. I don't think this is relevant anymore.

File: enterprise/internal/insights/background/pings/insights_ping_aggregators.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+	results := make([]types.OrgVisibleInsightPing, 0)
+	for rows.Next() {
+		var count int
+		var presentationType insightTypes.PresentationType
+		if err := rows.Scan(&presentationType, &count); err != nil {
+			return []types.OrgVisibleInsightPing{}, err
+		}
+
+		if presentationType == insightTypes.Line {
+			results = append(results, types.OrgVisibleInsightPing{Type: "search", TotalCount: count})
```
Comment (coury-clark) [inline enterprise/internal/insights/background/pings/insights_ping_aggregators.go:107]: This is fine, but eventually this is going to break when we break the variant that search insights -> line charts. Do you have any ideas on how we can ensure this doesn't just emit bad data when that happens if we forget?

**Review by coury-clark — APPROVED** [review]
Nice!

File: enterprise/internal/insights/background/pings/insights_ping_aggregators.go (hunk 3 above)
Comment (CristinaBirkel) [reply enterprise/internal/insights/background/pings/insights_ping_aggregators.go:107]: I thought about adding a comment here, but I'm not sure it will be noticed when modifying other parts of the code. When this changes, we'll still need some way of determining a search insight vs a langstats insight. Right now I think there are only two ways:
1. Pie vs Line
2. Whether or not the series has a non-zero `sample_interval_value`

I'm not sure either of these are dependable over the long term. And I'm not sure what the new way of determining this will be, so it's hard to know where to put that reminder.

Okay I made a sort of placeholder ticket for whenever we do this work. We could still miss it, but I think it will be less likely. https://github.com/sourcegraph/sourcegraph/issues/28478. Let me know if you have a better idea though!

File: enterprise/internal/insights/background/pings/insights_ping_aggregators.go (hunk 2 above)
Comment (CristinaBirkel) [reply enterprise/internal/insights/background/pings/insights_ping_aggregators.go:181]: Good call!
